modules/Microsoft.CDN/profiles/endpoints/origins/deploy.bicep

```diff
@@ -52,14 +64,22 @@ resource endpoint 'Microsoft.Cdn/profiles/endpoints@2021-06-01' existing = {
 resource origins 'Microsoft.Cdn/profiles/endpoints/origins@2021-06-01' = {
   parent: endpoint
   name: name
-  properties: {
-    hostName: hostName
-    httpPort: httpPort
-    enabled: enabled
-    httpsPort: httpsPort
-    priority: priority
-    weight: weight
-  }
+  properties: union({
+      hostName: hostName
+      httpPort: httpPort
+      enabled: enabled
+      httpsPort: httpsPort
+    }, ((priority > 0 || weight > 0) ? {
+      priority: priority
+      weight: weight
+    } : {}), (!empty(privateLinkAlias) && !empty(privateLinkLocation) ? {
+      privateLinkAlias: privateLinkAlias
+      privateLinkLocation: privateLinkLocation
+    } : {}), (!empty(privateLinkResourceId) ? {
```

Comment:
Aside from weight & priority - are these all properties that cannot have an empty/null value?

Reply:
Keep in mind, we should only use `union` if we absolutely have to.

Reply:
Correct - They do not support null/empty and I have failed deployment attempts trying this first